wagtail/wagtailadmin/messages.py

```diff
@@ -31,5 +33,32 @@ def error(request, message, buttons=None):
     return messages.error(request, render(message, buttons))
 
 
+def validation_error(request, message, form, buttons=None):
+    if not form.non_field_errors():
+        # just output the generic "there were validation errors" message, and leave
+        # the per-field highlighting to do the rest
+        full_message = message
+    else:
+        # display the full list of field and non-field validation errors
+        all_errors = []
+        for field_name, errors in form.errors.items():
+            if field_name == NON_FIELD_ERRORS:
+                prefix = ''
+            else:
+                try:
+                    field_label = form[field_name].label
+                except KeyError:
+                    field_label = field_name
+                prefix = "%s: " % field_label
+
+            for error in errors:
+                all_errors.append(prefix + error)
+
+        errors_html = format_html_join('\n', '<li>{}</li>', ((e,) for e in all_errors))
+        full_message = format_html("""{} <ul class="errorlist">{}</ul>""", message, errors_html)
+
+    return messages.error(request, render(full_message, buttons))
```

Comment:
I am not sure if this will occur in a real-life situation, but the format of the `full_message` for `NON_FIELD_ERRORS` can cause a visual conflict if they also include buttons. At this point it will render the main `ul` and the nested `.errorlist` which causes the buttons to end up at the bottom of the list. 

For visual consistency it would probably make more sense if they are next to the main message followed by the nested `.errorlist`. 

Example from the styleguide:
![image](https://cloud.githubusercontent.com/assets/287422/23834504/de7c9d2e-0757-11e7-8224-eb4b353a474f.png)